src/mscorlib/src/System/String.cs

```diff
@@ -851,14 +851,12 @@ private unsafe String CtorCharPtrStartLength(char *ptr, int startIndex, int leng
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         internal extern bool TryGetTrailByte(out byte data);
 
-#if !FEATURE_CORECLR
         public CharEnumerator GetEnumerator() {
             Contract.Ensures(Contract.Result<CharEnumerator>() != null);
             Contract.EndContractBlock();
             BCLDebug.Perf(false, "Avoid using String's CharEnumerator until C# special cases foreach on String - use the indexed property on String instead.");
```

Comment:
I also thought it might be like an assert, if you see we also have it in some other already public members like line 864 or 872 of this same file, so I doubt that we have to do anything special. @weshaggard do you know if we have to remove that line?